frontend/src/components/Avatar/Avatar.tsx

```diff
@@ -0,0 +1,76 @@
+import {
+  Avatar as ChakraAvatar,
+  AvatarBadge as ChakraAvatarBadge,
+  AvatarProps as ChakraAvatarProps,
+  Box,
+  ButtonProps,
+  Icon,
+  MenuItemProps,
+  useMultiStyleConfig,
+} from '@chakra-ui/react'
+
+import { BxsUser } from '~/assets/icons/BxsUser'
+
+import DropdownMenu from '../Menu'
+
+export interface AvatarProps extends ChakraAvatarProps {
+  hasNotification?: boolean
+}
+
+/*
+ * Utility to extract first letter of avatar name
+ * Override default getInitials Avatar method
+ */
+const getInitials = (fullName: string) => fullName[0]
+
+/*
+ * Avatar component
+ * May be wrapped in AvatarMenuButton
+ */
+export const Avatar = (props: AvatarProps): JSX.Element => {
+  const avatarBadge = props.hasNotification ? <ChakraAvatarBadge /> : ''
+
+  return (
+    <ChakraAvatar getInitials={getInitials} {...props}>
+      {avatarBadge}
+    </ChakraAvatar>
+  )
+}
+
+/**
+ * DropdownMenuButton styled for avatar
+ * Used to wrap Avatar component
+ * @preconditions Must be a child of DropdownMenu component,
+ * and returned using a render prop (see implementation in Avatar.stories).
+ */
+const AvatarMenuButton = (props: ButtonProps): JSX.Element => {
```

Comment:
as above, I don't think this belongs in the Avatar design component, but maybe we can create a template that uses Avatar component, with separate stories